component/cluster.libsonnet

```diff
@@ -0,0 +1,339 @@
+  local sa = kube.ServiceAccount('vc-' + name) {
+    metadata+: {
+      namespace: options.namespace,
+    },
+  };
+  local role = kube.Role(name) {
+    metadata+: {
+      namespace: options.namespace,
+    },
+    rules: [
+      {
+        apiGroups: [
+          '',
+        ],
+        resources: [
+          'configmaps',
+          'secrets',
+          'services',
+          'pods',
+          'pods/attach',
+          'pods/portforward',
+          'pods/exec',
+          'endpoints',
+          'persistentvolumeclaims',
+        ],
+        verbs: [
+          'create',
+          'delete',
+          'patch',
+          'update',
+          'get',
+          'list',
+          'watch',
+        ],
+      },
+      {
+        apiGroups: [
+          '',
+        ],
+        resources: [
+          'events',
+          'pods/log',
+        ],
+        verbs: [
+          'get',
+          'list',
+          'watch',
+        ],
+      },
+      {
+        apiGroups: [
+          'networking.k8s.io',
+        ],
+        resources: [
+          'ingresses',
+        ],
+        verbs: [
+          'create',
+          'delete',
+          'patch',
+          'update',
+          'get',
+          'list',
+          'watch',
+        ],
+      },
+      {
+        apiGroups: [
+          'apps',
+        ],
+        resources: [
+          'statefulsets',
+          'replicasets',
+          'deployments',
+        ],
+        verbs: [
+          'get',
+          'list',
+          'watch',
+        ],
+      },
+    ] + if isOpenshift then
+      [
+        {
+          apiGroups: [
+            '',
+          ],
+          resources: [
+            'endpoints/restricted',
+          ],
+          verbs: [
+            'create',
+          ],
+        },
+      ] else [],
+  };
+  local roleBinding = kube.RoleBinding(name) {
+    metadata+: {
+      namespace: options.namespace,
+    },
+    subjects_: [ sa ],
+    roleRef_: role,
+  };
+
+  local service = kube.Service(name) {
+    metadata+: {
+      namespace: options.namespace,
+    },
+    spec: {
+      type: 'ClusterIP',
+      ports: [
+        {
+          name: 'https',
+          port: 443,
+          targetPort: 8443,
+          protocol: 'TCP',
+        },
+      ],
+      selector: {
+        app: 'vcluster',
+        release: name,
+      },
+    },
+  };
+
+  local headlessService = kube.Service(name + '-headless') {
+    metadata+: {
+      namespace: options.namespace,
+    },
+    spec: {
+      ports: [
+        {
+          name: 'https',
+          port: 443,
+          targetPort: 8443,
+          protocol: 'TCP',
+        },
+      ],
+      clusterIP: 'None',
+      selector: {
+        app: 'vcluster',
+        release: name,
+      },
+    },
+  };
+
+  local statefulSet = kube.StatefulSet(name) {
+    metadata+: {
+      namespace: options.namespace,
+    },
+    spec: {
+      serviceName: name + '-headless',
+      replicas: 1,
```

Comment:
How can we have achieve HA of the control-api?

Reply:
We'd need to switch to vanilla k8s vcluster. Which brings a huge amount of complexity.

Reply:
Then we'll need to check with @corvus-ch  if non-HA is acceptable.

Reply:
The choice was made to go with that vcluster solution. So this is what we get. I guess @tobru and I have to think about this but this is out of scope for this PR.

Reply:
Yes, this is out-of-scope in this implementation. Having 1 replica is good enough for now. We already have an underlying Kubernetes cluster which takes care of the vcluster.